js/components/common/__tests__/CourseModalButtons.test.js

```diff
@@ -0,0 +1,75 @@
+import { screen, waitFor, fireEvent } from "@testing-library/react"
+import TestGraph from "../../graph/__tests__/TestGraph.js"
+
+describe("CourseModal", () => {
+  beforeEach(async () => {
+    await TestGraph.build()
+
+    // find the course node AAA100 and open its info modal
+    const node = document.querySelector('[data-testid="aaa100"]')
+    fireEvent.mouseOver(node)
+    const infobox = document.getElementById("aaa100-tooltip-rect")
+    fireEvent.click(infobox)
+  })
+
+  // wait for the modal to appear
+  const waitForModalUpdate = async () => {
+    await waitFor(() => {
+      const updatedModal = document.querySelector(".ReactModalPortal")
```

Comment:
Since you don't actually need the return value (other than to check falsiness for `waitFor`), you can simplify this to just `waitFor(() => document.querySelector(...))`